setup.py

```diff
@@ -28,6 +28,10 @@
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ###############################################################################
 
+import sys
+if (sys.version_info < (3, 5)):
+    raise Exception('cocotb requires Python version >= 3.5')
```

Comment:
Could follow `numpy`'s lead here: https://github.com/numpy/numpy/blob/master/setup.py#L29-L30
```suggestion
    raise RuntimeError('cocotb requires Python version >= 3.5')
```